File: backend/src/main/kotlin/com/example/backend/models/Concerner.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "concerner")
class Concerner {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "vente_id")
  var venteId: Long? = null

  @Column(name = "produit_id")
  var produitId: Int? = null

  @Column(name = "en_rayon_id")
  var enRayonId: String? = null

  @Column(name = "prixUnit")
  var prixUnit: Int? = null

  @Column(name = "quantite")
  var quantite: Int? = null

  @Column(name = "reduction")
  var reduction: Int? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var supprimer: Int? = 0

  @ColumnDefault("'en rayon'")
  @Column(name = "type")
  var type: String? = null
}

File: backend_py/pharmaPython/app/models/concerner.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship

from app.core.db import Base


class Concerner(Base):
  __tablename__ = "concerner"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, index=True)
  vente_id = Column(Integer, ForeignKey("vente.id"))
  # vente_id = Column(Integer)
  produit_id = Column(Integer)
  en_rayon_id = Column(String)
  prix_unit = Column(Integer, default=0)
  quantite = Column(Integer, default=0)
  reduction = Column(Integer, default=0)
  supprimer = Column(Integer, default=0)
  type = Column(String, default="en rayon")

  vente = relationship("Vente")
